geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/RasterRDD.scala

```diff
@@ -90,74 +103,86 @@ abstract class RasterRDD[K: AvroRecordCodec: ClassTag] extends TileRDD[K] {
-
-  def collectMetadata(crs: Option[CRS], extent: Option[Extent], layout: Option[TileLayout], tileSize: Int): String = {
-      (crs, extent, layout) match {
-        case (Some(crs), Some(extent), Some(layout)) =>
-          rdd.collectMetadata[SpaceTimeKey](crs, LayoutDefinition(extent, layout))
-        case (None, Some(extent), Some(layout)) =>
-          rdd.collectMetadata[SpaceTimeKey](LayoutDefinition(extent, layout))
-        case (Some(crs), _, _) =>
-          rdd.collectMetadata[SpaceTimeKey](crs, FloatingLayoutScheme(tileSize, tileSize))._2
-        case (None, None, None) =>
-          rdd.collectMetadata[SpaceTimeKey](FloatingLayoutScheme())._2
-        case _ =>
-          throw new IllegalArgumentException(s"Can't handle $crs, $extent, $layout, $tileSize")
-      }
+class TemporalRasterRDD(val rdd: RDD[(TemporalProjectedExtent, MultibandTile)]) extends RasterRDD[TemporalProjectedExtent] {
+
+  def collectMetadata(layout: Either[LayoutScheme, LayoutDefinition], crs: String): String = {
+    (TileRDD.getCRS(crs), layout) match {
+      case (Some(crs), Right(layoutDefinition)) =>
+          rdd.collectMetadata[SpaceTimeKey](crs, layoutDefinition)
+      case (None, Right(layoutDefinition)) =>
+          rdd.collectMetadata[SpaceTimeKey](layoutDefinition)
+      case (Some(crs), Left(layoutScheme)) =>
+          rdd.collectMetadata[SpaceTimeKey](crs, layoutScheme)._2
+      case (None, Left(layoutScheme)) =>
+          rdd.collectMetadata[SpaceTimeKey](layoutScheme)._2
+    }
   }.toJson.compactPrint
 
   def cutTiles(layerMetadata: String, resampleMethod: String): TiledRasterRDD[SpaceTimeKey] = {
     val md = layerMetadata.parseJson.convertTo[TileLayerMetadata[SpaceTimeKey]]
     val rm = TileRDD.getResampleMethod(resampleMethod)
     val tiles = rdd.cutTiles[SpaceTimeKey](md, rm)
-    new TemporalTiledRasterRDD(MultibandTileLayerRDD(tiles, md))
+    new TemporalTiledRasterRDD(None, MultibandTileLayerRDD(tiles, md))
   }
 
   def tileToLayout(layerMetadata: String, resampleMethod: String): TiledRasterRDD[SpaceTimeKey] = {
     val md = layerMetadata.parseJson.convertTo[TileLayerMetadata[SpaceTimeKey]]
     val rm = TileRDD.getResampleMethod(resampleMethod)
-    new TemporalTiledRasterRDD(MultibandTileLayerRDD(rdd.tileToLayout(md, rm), md))
+    new TemporalTiledRasterRDD(None, MultibandTileLayerRDD(rdd.tileToLayout(md, rm), md))
   }
 
-  def reproject(target_crs: String, resampleMethod: String): TemporalProjectedRasterRDD = {
-    val tcrs = TileRDD.getCRS(target_crs).get
+  def reproject(targetCRS: String, resampleMethod: String): TemporalRasterRDD = {
+    val crs = TileRDD.getCRS(targetCRS).get
     val resample = TileRDD.getResampleMethod(resampleMethod)
-    new TemporalProjectedRasterRDD(rdd.reproject(tcrs, resample))
+    new TemporalRasterRDD(rdd.reproject(crs, resample))
   }
 }
+
+object ProjectedRasterRDD {
+  def apply(javaRDD: JavaRDD[Array[Byte]], schema: String): ProjectedRasterRDD =
```

Comment:
These functions make no sense from python side since they will be unfamiliar and you can't use the scala apply syntax sugar. `fromAvroEncodedRDD`